packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx

```diff
@@ -50,10 +79,82 @@ export default class IcicleEventChart extends Component<Props> {
   }
 
   // Check for changed data to rerender the icicle chart
-  componentDidUpdate(prevProps: Props) {}
+  componentDidUpdate(prevProps: Props) {
+    const root = this.props.d3TreeRoot;
+    const prevRoot = prevProps.d3TreeRoot;
+
+    if (root.data.id !== prevRoot.data.id || root.data.value !== prevRoot.data.value) {
+      this.renderIcicleChart();
+    }
+  }
 
   // Creates chart using svg & chartRef to the div element
-  renderIcicleChart() {}
+  renderIcicleChart() {
+    const { boxMargin, color, contentRenderer, isVertical, width, height, rounding } = this.props;
+
+    const root = this.props.d3TreeRoot;
+
+    // Clear all elements and redraw the Icicle Chart
+    d3Select(this.chartRef.current)
+      .selectAll('*')
+      .remove();
+
+    const svg = d3Select(this.chartRef.current)
```

Comment:
I'm curious why you aren't using a library like [`@vx` ](https://vx-demo.now.sh/gallery) that plays better with `react` and renders visualizations/utilizes `d3` more declaratively.

Reply:
I have no good answer for that other than being not very familiar with vx... I recall seeing it in the `event-flow` viz if I remember correctly but didn't look too much into it. Will definitely look more into that library tomorrow in the morning!

Reply:
Very good point! The visualization will have some animation as well and it doesn't look like vx supports it. Although we could use other react animation libraries with vx. I think given Chris's deadlines, I suggest that this is something we look into later on. I am not a big fan of how unreadable d3 either but fortunately, I don't think there is much more than what is already in this PR.